### PR TITLE
feat(styleguide): Use file for story hierarchy 

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Makes story hierarchy dynamically derived based on our folder structure using a babel macro. 

* Adds `@codecademy/macros` as npm module for simple babel macros.
* Adds a macro for extracting the correct design system heirarchy from the current file path.
* Moves stories from unnecessary nesting that is not reflective of storybook hierarchy.
* Changes stories in their own directories for other helpers to be named `index` and have that path ignored when generating the title.
* Updates storyShot splitter config to determine the total number of chunks based on file count (we can add more as needed) without updates to variable counts.

#### Usage 
Named files: `stories/Atoms/Button.stories.mdx`
```tsx
import title from '@codecademy/macros/lib/title.macro';

<Meta title={title} /> // title => 'Atoms/Button' 
```
Index files:  `stories/Layouts/Box/index.stories.mdx`
```tsx
import title from '@codecademy/macros/lib/title.macro';

<Meta title={title} /> // title => 'Layouts/Box' 
```

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60006eb3b626db0b212a8845--gamut-preview.netlify.app>https://60006eb3b626db0b212a8845--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60006eb3b626db0b212a8845>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.fc73cb.0<br/>@codecademy/styleguide@21.9.1-alpha.fc73cb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001bc41e49a45593c8aa210--gamut-preview.netlify.app>https://6001bc41e49a45593c8aa210--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001bc41e49a45593c8aa210>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.fa0229.0<br/>@codecademy/styleguide@21.9.1-alpha.fa0229.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001c2ca824ad62d7a83374d--gamut-preview.netlify.app>https://6001c2ca824ad62d7a83374d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001c2ca824ad62d7a83374d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.622d2f.0<br/>@codecademy/styleguide@21.9.1-alpha.622d2f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001c4e95fd9543e922dfdb7--gamut-preview.netlify.app>https://6001c4e95fd9543e922dfdb7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001c4e95fd9543e922dfdb7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.9196ff.0<br/>@codecademy/styleguide@21.9.1-alpha.9196ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001c7f4f0a0cb5b49c797dd--gamut-preview.netlify.app>https://6001c7f4f0a0cb5b49c797dd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001c7f4f0a0cb5b49c797dd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.a6e2da.0<br/>@codecademy/styleguide@21.9.1-alpha.a6e2da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001cb0410c9893e05a38e1a--gamut-preview.netlify.app>https://6001cb0410c9893e05a38e1a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001cb0410c9893e05a38e1a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001d3fc301e10602606eaac--gamut-preview.netlify.app>https://6001d3fc301e10602606eaac--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001d3fc301e10602606eaac>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.3a922c.0<br/>@codecademy/styleguide@21.9.1-alpha.3a922c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001d7bc0c48c364f402d894--gamut-preview.netlify.app>https://6001d7bc0c48c364f402d894--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001d7bc0c48c364f402d894>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.84c1eb.0<br/>@codecademy/styleguide@21.9.1-alpha.84c1eb.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.21bd5b.0<br/>@codecademy/styleguide@21.9.1-alpha.21bd5b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.ca8692.0<br/>@codecademy/styleguide@21.9.1-alpha.ca8692.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001e8480c48c37c7a02d92e--gamut-preview.netlify.app>https://6001e8480c48c37c7a02d92e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001e8480c48c37c7a02d92e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.7b90ba.0<br/>@codecademy/styleguide@21.9.1-alpha.7b90ba.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.3f9848.0<br/>@codecademy/styleguide@21.9.1-alpha.3f9848.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001eca1914888008b7e582c--gamut-preview.netlify.app>https://6001eca1914888008b7e582c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001eca1914888008b7e582c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.888459.0<br/>@codecademy/styleguide@21.9.1-alpha.888459.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001ef88eb42f60138d32a13--gamut-preview.netlify.app>https://6001ef88eb42f60138d32a13--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001ef88eb42f60138d32a13>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Dope. You should write a blog post or give a demo or something for this. 😄

File: packages/macros/lib/title.macro.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const absolute = path.resolve(f).split('/');
+  const storyRoot = 'stories';
+
+  const storyPath = absolute.slice(
+    absolute.indexOf(storyRoot) + 1,
+    absolute.length
+  );
+
+  const title = storyPath
+    .filter((slug) => slug.indexOf('index') === -1 && slug.length > 0)
```
Comment (JoshuaKGoldberg) [inline packages/macros/lib/title.macro.js:16]: ⛳ ✨ 

```suggestion
    .filter((slug) => slug && !slug.contains('index'))
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.c2874b.0<br/>@codecademy/styleguide@21.13.1-alpha.c2874b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001f1eb55153f0132e426d5--gamut-preview.netlify.app>https://6001f1eb55153f0132e426d5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001f1eb55153f0132e426d5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.edd1d0.0<br/>@codecademy/styleguide@21.13.1-alpha.edd1d0.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.48b912.0<br/>@codecademy/styleguide@21.13.1-alpha.48b912.0<br/></pre>

File: packages/macros/lib/title.macro.js (hunk 1 above)
Comment (codecaaron) [reply packages/macros/lib/title.macro.js:16]: Gonna make a note to transpile this so I can get access to nicer string methods.  <3

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001f6a35d15c300de29cb78--gamut-preview.netlify.app>https://6001f6a35d15c300de29cb78--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001f6a35d15c300de29cb78>Deploy Logs</a></p>

File: packages/macros/lib/title.macro.js (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/macros/lib/title.macro.js:16]: Oh right, older Node versions. Ignore me 😄

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.03fdf2.0<br/>@codecademy/styleguide@21.13.1-alpha.03fdf2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001fdaacc9a420bd0f23b74--gamut-preview.netlify.app>https://6001fdaacc9a420bd0f23b74--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001fdaacc9a420bd0f23b74>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.e4c7d4.0<br/>@codecademy/styleguide@21.13.1-alpha.e4c7d4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6002000be7d51e0bc36e4e44--gamut-preview.netlify.app>https://6002000be7d51e0bc36e4e44--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6002000be7d51e0bc36e4e44>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.2e399c.0<br/>@codecademy/styleguide@21.13.1-alpha.2e399c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600202515d15c30c8c29cb5a--gamut-preview.netlify.app>https://600202515d15c30c8c29cb5a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600202515d15c30c8c29cb5a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.095841.0<br/>@codecademy/styleguide@21.13.1-alpha.095841.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600205b8ef91db0ea4bd99cc--gamut-preview.netlify.app>https://600205b8ef91db0ea4bd99cc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600205b8ef91db0ea4bd99cc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.9f193e.0<br/>@codecademy/styleguide@21.13.1-alpha.9f193e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6002088655153f1a45e425e3--gamut-preview.netlify.app>https://6002088655153f1a45e425e3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6002088655153f1a45e425e3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.aa4df6.0<br/>@codecademy/styleguide@21.13.1-alpha.aa4df6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60020b1301aff100afdc71d8--gamut-preview.netlify.app>https://60020b1301aff100afdc71d8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60020b1301aff100afdc71d8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.0a3ffe.0<br/>@codecademy/styleguide@21.13.1-alpha.0a3ffe.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6002222e2227c41165514baf--gamut-preview.netlify.app>https://6002222e2227c41165514baf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6002222e2227c41165514baf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.5b3e27.0<br/>@codecademy/styleguide@21.13.1-alpha.5b3e27.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60022545fe106c10670a3d69--gamut-preview.netlify.app>https://60022545fe106c10670a3d69--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60022545fe106c10670a3d69>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.85ae9c.0<br/>@codecademy/styleguide@21.13.1-alpha.85ae9c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6002295d630b6b2c28f72342--gamut-preview.netlify.app>https://6002295d630b6b2c28f72342--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6002295d630b6b2c28f72342>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.e93d28.0<br/>@codecademy/styleguide@21.13.2-alpha.e93d28.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@1.0.1-alpha.9dbb4e.0<br/>@codecademy/styleguide@21.13.2-alpha.9dbb4e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60022c8618eb62205e329011--gamut-preview.netlify.app>https://60022c8618eb62205e329011--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60022c8618eb62205e329011>Deploy Logs</a></p>

**Pull request merged**